torchmetrics/classification/binned_precision_recall.py

```diff
@@ -118,14 +128,26 @@ def __init__(
         )
 
         self.num_classes = num_classes
-        self.num_thresholds = num_thresholds
-        thresholds = torch.linspace(0, 1.0, num_thresholds)
-        self.register_buffer("thresholds", thresholds)
+        if num_thresholds is not None:
+            if thresholds is not None:
+                raise ValueError(
+                    'Arguments `num_thresholds` and `thresholds` are mutually exclusive, but got'
+                    f'{num_thresholds} and {thresholds}'
+                )
+            self.num_thresholds = num_thresholds
+            thresholds = torch.linspace(0, 1.0, num_thresholds)
+            self.register_buffer("thresholds", thresholds)
+        elif thresholds is not None:
```

Comment:
I would switch as to use `thresholds` first, with `num_thresholds` default 100 it will be confusing

Reply:
what do you think about deprecating the `num_thresholds` argument and letting `thresholds` also be an int (so it basically will cover both the old and new behaviour) ?

Reply:
well yes, that would be probably the best :]